akka-docs/src/main/paradox/stream/operators/RetryFlow/withBackoff.md

```diff
@@ -0,0 +1,54 @@
+# RetryFlow.withBackoff
+
+Wrap the given @apidoc[Flow] and retry individual elements in that stream with an exponential backoff. A decider function tests every emitted element and can return a new element to be sent to the wrapped flow for another try.
+
+@ref[Error handling](../index.md#error-handling)
+
+## Signature
+
+Scala
+:   @@signature [RetryFlow.scala](/akka-stream/src/main/scala/akka/stream/scaladsl/RetryFlow.scala) { #withBackoff }
+
+Java
+:   @@snip [RetryFlowTest.java](/akka-stream-tests/src/test/java/akka/stream/javadsl/RetryFlowTest.java) { #withBackoff-signature }
+
+
+## Description
+
+When an element is emitted by the wrapped `flow` it is passed to the `decideRetry` function, which may return an element to retry in the `flow`. 
+
+The retry backoff is controlled by the `minBackoff`, `maxBackoff` and `randomFactor` parameters.
+At most `maxRetries` will be made after the initial try.
+
+The wrapped `flow` must have **one-in one-out semantics**. It may not filter, nor duplicate elements. The `RetryFlow` will fail if two elements are emitted from the `flow`, it will be stuck "forever" if nothing is emitted. Just one element will be emitted into the `flow` at any time. The `flow` needs to emit an element before the next will be emitted to it. 
```

Comment:
```suggestion
The wrapped `flow` must have **one-in one-out semantics**. It may not filter, nor duplicate or reorder elements. The `RetryFlow` will fail if two elements are emitted from the `flow`, it will be stuck "forever" if nothing is emitted. Just one element will be emitted into the `flow` at any time. The `flow` needs to emit an element before the next will be emitted to it. 
```

Reply:
With these restrictions, wouldn't it make more sense to have a `retryMapAsync` rather than `retryFlow`?

I guess the same reasoning goes for `FlowWithContext` in general, that it's nice to be able to use our streams building blocks?

Reply:
The retrying is most relevant with access to external services which often involve materialising a client or connection. Using flows, those can be shared across all requests.

Reply:
BTW Reording is impossible, as it will only receive one element at a time.